src/ArrayOfArrays.hpp

```diff
@@ -178,6 +184,18 @@ class ArrayOfArrays : protected ArrayOfArraysView< T, INDEX_TYPE >
   void resize( INDEX_TYPE const numSubArrays, INDEX_TYPE const defaultArrayCapacity=0 )
   { ParentClass::resize( numSubArrays, defaultArrayCapacity ); }
 
+  /**
+   * @tparam POLICY The RAJA policy used to convert @p capacities into the offsets array.
+   *   Should NOT be a device policy.
+   * @brief Clears the array and creates a new array with the given number of sub-arrays.
+   * @param numSubArrays The new number of arrays.
+   * @param capacities A pointer to an array of length @p numSubArrays containing the capacity
+   *   of each new sub array.
+   */
+  template< typename POLICY >
+  void resizeFromCapacities( INDEX_TYPE const numSubArrays, INDEX_TYPE const * const capacities )
```

Comment:
This allows you to pre-calculate the capacities of the array and resize efficiently.